core/src/main/java/com/lafaspot/sparzr/InternalSaxHandler.java

```diff
@@ -0,0 +1,355 @@
+                // The internal json-ld element is not well formed. So let's skip this potential json schema.org and continue with other
+                // schema.org in the same html. Just clean up and move on. No need to report it. Also, if we throw the SAXException, from
+                // here it will just stop processing the remaining document. Not a good idea. We still want to extract whatever it well
+                // formed.
+                containsJsonld = false;
+                stringBuffer = new StringBuffer();
+                tagTypes.pop();
+                return;
+            } catch (IOException e) {
+                throw new SAXException("I/O failed.", e);
+            }
+            parseJson(element);
+            for (final Listener listener : listeners) {
+                listener.foundItem(element);
+            }
+            containsJsonld = false;
+            stringBuffer = new StringBuffer();
+            tagTypes.pop();
+        } else {
+            /* Else, return top-level items if possible (Microdata) */
+            ObjectNode currentItem = null;
+            if (!items.isEmpty()) {
+                currentItem = items.getLast();
+            }
+            // Get property content from in-between tags if required
+            if (propertyName != null && currentItem != null) {
+                final String content = stringBuffer.toString().trim();
+                if (currentItem.get(propertyName) != null && currentItem.get(propertyName).isArray()) {
+                    ArrayNode jArray = (ArrayNode) currentItem.get(propertyName);
+                    jArray.add(new TextNode(content));
+                } else {
+                    final ArrayNode itemArray = jsonNodeFactory.arrayNode();
+                    itemArray.add(new TextNode(content));
+                    currentItem.set(propertyName, itemArray);
+                }
+                propertyName = null;
+                stringBuffer = new StringBuffer();
+            }
+            // If closing tag of an item, move up a level and return top-level item if necessary
+            final Format type = tagTypes.pop();
+            if (type.equals(Format.MICRODATA)) {
+                items.removeLast();
+                if (items.isEmpty() && currentItem != null) {
+                    for (final Listener listener : listeners) {
+                        listener.foundItem(currentItem);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Buffer characters from in-between tags.
+     */
+    @Override
+    public void characters(final char[] ch, final int start, final int length) throws SAXException {
+        if (containsJsonld || propertyName != null) {
+            stringBuffer.append(ch, start, length);
+        }
+    }
+
+    /**
+     * Recursively parse json to find all itemtypes.
+     *
+     * @param element Root json element
+     */
+    private void parseJson(final JsonNode element) {
```

Comment:
This will couple the interface to faster xml, see if that is what we want or make this string or string buffer, and if we are going to couple this library with faster xml parser, lets describe that in the package docs.